hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/Path.java

```diff
@@ -420,22 +421,37 @@ public String getName() {
   }
 
   /**
-   * Returns the parent of a path or null if at root.
+   * Returns the parent of a path or null if at root. Better alternative is
+   * {@link #getParentPath()} to handle nullable value for root path.
+   *
    * @return the parent of a path or null if at root
    */
   public Path getParent() {
+    return getParentUtil();
+  }
+
+  /**
+   * Returns the parent of a path as Optional or empty() if at root.
+   *
+   * @return Parent of path wrappen in Optional. empty() if at root.
```

Comment:
I prefer to explain `empty()` in words.
```suggestion
   * Returns the parent of a path as Optional or an empty Optional if at root.
   *
   * @return Parent of path wrapped in Optional. An empty Optional if at root.
```

Reply:
I think this is nice, perhaps I can directly give link to Optional and Optional#empty() as well.